plugins/de.cognicrypt.codegenerator.tests/src/de/cognicrypt/codegenerator/generator/test/QuestionsToTemplateTests.java

```diff
@@ -0,0 +1,544 @@
+		question1answer4.setOption("");
+		Answer question2answer4 = new Answer();
+		question2answer4.setValue("Other/Do not know");
+		question2answer4.setOption("");
+		HashMap<Question, Answer> constraints = new HashMap<Question, Answer>();
+		constraints.put(question1, question1answer4);
+		constraints.put(question2, question2answer4);
+		wizard.constructConstraints(constraints);
+		String expected = "src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryption";
+		assertEquals(wizard.constructTemplateName(), expected);
+	}
+	
+	@Test
+	public void testEncryption17() {
+		AltConfigWizard wizard = new AltConfigWizard();
+		Task encryption = TestUtils.getTask("Encryption");
+		wizard.setSelectedTask(encryption);
+		Question question1 = new Question();
+		question1.setId(0);
+        question1.setQuestionText("Which method of communication would you prefer to use for key exchange?");
+		Question question2 = new Question();
+		question2.setId(1);
+        question2.setQuestionText("What data type do you wish to encrypt?");
+		Answer question1answer5 = new Answer();
+		question1answer5.setValue("No Sharing");
+		question1answer5.setOption("");
+		Answer question2answer1 = new Answer();
+		question2answer1.setValue("Byte Array");
+		question2answer1.setOption("");
+		HashMap<Question, Answer> constraints = new HashMap<Question, Answer>();
+		constraints.put(question1, question1answer5);
+		constraints.put(question2, question2answer1);
+		wizard.constructConstraints(constraints);
+		String expected = "src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryption";
+		assertEquals(wizard.constructTemplateName(), expected);
+	}
+	
+	@Test
+	public void testEncryption18() {
+		AltConfigWizard wizard = new AltConfigWizard();
+		Task encryption = TestUtils.getTask("Encryption");
+		wizard.setSelectedTask(encryption);
+		Question question1 = new Question();
+		question1.setId(0);
+        question1.setQuestionText("Which method of communication would you prefer to use for key exchange?");
+		Question question2 = new Question();
+		question2.setId(1);
+        question2.setQuestionText("What data type do you wish to encrypt?");
+        Answer question1answer5 = new Answer();
+		question1answer5.setValue("No Sharing");
+		question1answer5.setOption("");
+		Answer question2answer2 = new Answer();
+		question2answer2.setValue("File");
+		question2answer2.setOption("files");
+		HashMap<Question, Answer> constraints = new HashMap<Question, Answer>();
+		constraints.put(question1, question1answer5);
+		constraints.put(question2, question2answer2);
+		wizard.constructConstraints(constraints);
+		String expected = "src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryptionfiles";
+		assertEquals(wizard.constructTemplateName(), expected);
+	}
+	
+	@Test
+	public void testEncryption19() {
+		AltConfigWizard wizard = new AltConfigWizard();
+		Task encryption = TestUtils.getTask("Encryption");
+		wizard.setSelectedTask(encryption);
```

Comment:
It looks like a lot of duplicate code, althogh I understand it is not. Is it possible to parametrize the construction of the wizard to reduce the lines of code. Ignore this feedback if the parametrization would make the construction of wizard call unduly lengthy and ugly.

Reply:
It can also be done. Thanks for the suggestion :)